modules/ding_campaign_plus/modules/ding_campaign_plus_facet/ding_campaign_plus_facet.module

```diff
@@ -0,0 +1,1094 @@
+      '#type' => 'select',
+      '#title' => t('Default common number'),
+      '#description' => t('Default "Most common values" to match against.'),
+      '#options' => drupal_map_assoc(range(1, 30)),
+      '#default_value' => empty($default['default']) ? '_none_' : $default['default'],
+    ),
+    'facets' =>  array(
+      '#type' => 'radios',
+      '#title' => t('Facets available'),
+      '#description' => t('The contents subjects will automatically create a trigger against the selected facet type.'),
+      '#options' => array('_none_' => t('None')) + $options,
+      '#default_value' => empty($default['facets']) ? '_none_' : $default['facets'],
+    ),
+  );
+}
+
+/**
+ * Implements hook_form_ding_campaign_plus_admin_settings_alter().
+ *
+ * Added form elements to the configuration form to define which facets should
+ * be available for facet triggers.
+ */
+function ding_campaign_plus_facet_form_ding_campaign_plus_admin_settings_alter(&$form, &$form_state, $form_id) {
+  $form['ding_campaign_plus_facets'] = array(
+    '#type' => 'fieldset',
+    '#title' => 'Facets',
+    '#tree' => TRUE,
+    '#collapsible' => TRUE,
+    '#collapsed' => TRUE,
+    '#weight' => 0,
+  );
+
+  $form['ding_campaign_plus_facets']['facets'] = array(
+    '#prefix' => '<div id="facets">',
+    '#suffix' => '</div>',
+    '#theme' => 'ding_campaign_plus_facet_table_drag',
+  );
+
+  $id = 0;
+  $facets = _ding_campaign_plus_facet_get_facets();
+  foreach ($facets as $id => $facet) {
+    $form['ding_campaign_plus_facets']['facets'][$id]['facet'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Facet id'),
+      '#title_display' => 'invisible',
+      '#default_value' => $facet['facet'],
+    );
+
+    $form['ding_campaign_plus_facets']['facets'][$id]['title'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Title'),
+      '#title_display' => 'invisible',
+      '#default_value' => $facet['title'],
+    );
+
+    $form['ding_campaign_plus_facets']['facets'][$id]['action'] = array(
+      '#type' => 'link',
+      '#title' => t('delete'),
+      '#href' =>  'admin/config/ding/campaigns_plus/' . $facet['facet'] . '/delete',
+      '#options' => array('attributes' => array('title' => t('Delete facet.'))),
+    );
+  }
+
+  $id++;
+  $form['ding_campaign_plus_facets']['facets'][$id]['facet'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Facet id'),
+    '#default_value' => '',
+  );
+
+  $form['ding_campaign_plus_facets']['facets'][$id]['title'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Title'),
+    '#default_value' => '',
+  );
+
+  $form['ding_campaign_plus_facets']['description'] = array(
+    '#prefix' => '<div class="description">',
+    '#suffix' => '</div>',
+    '#markup' => t('All facets are explained at <a href="!url">!url</a>', array('!url' => 'https://opensearch.addi.dk/b3.5_4.5/?showCqlFile&repository=prod&cql=opensearch_cql.xml#facet')),
+  );
+
+  $form += array('#submit' => array());
```

Comment:
this is redundant

Reply:
If not done this way submit handlers for other parts of the form may be overridden of if this is the first of the sub-modules it may not exists yet.

Reply:
I beg to differ

Reply:
okay, if you are sure... i'll remove it